### ISSUE TITLE
build_ami shouldn't do "rm -rf build/*"

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#4189]
The script dist/ami/build_ami.sh has:

```sh
if [ $LOCALRPM -eq 1 ]; then
    sudo rm -rf build/*
    ...
    if [ ! -f dist/ami/files/scylla.x86_64.rpm ]
         ...
        reloc/build_reloc.sh
```

The "rm -rf build" (and by the way, why sudo?) means that even if the user already compiled Scylla, this will remove everything and recompile. This recompilation won't take long if you use ccache, but it still unnecessary.

The recompilation is unnecessary, because all build/reloc.sh does to compile is to run "ninja", which is presumably exactly what the user did earlier, so why delete whatever partial compilation work already done? Moreover, even for a first build (without a primed ccache), the user should probably build **before** running build_ami because he knows of a faster way to build (e.g., using distcc) that the script doesn't.

Finally, if you ever run build_ami.sh a **second** time, this code will remove everything in build/, but then not rebuild anything (because scylla.x86_64.rpm already exists), so you'll be left with an empty build directory. This is annoying because the user can no longer run Scylla or tests or whatever he wanted to have in the build directory. Again, I can run "ninja" again and quickly (if I use ccache) get my build directory back - but it's still annoying.

**Comment by syuu1228** [comment]
@nyh You are correct, patch sent on ML: "[PATCH scylla] dist/ami: don't cleanup build dir"
Thanks pointing out the issue.
